profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc

```diff
@@ -24,6 +24,67 @@ function nexteuropa_varnish_admin_settings_form($form, $form_state) {
     '#default_value' => variable_get('nexteuropa_varnish_default_purge_rule', FALSE),
   );
 
+  $form = system_settings_form($form);
+  array_unshift($form['#submit'], '_nexteuropa_varnish_default_rule_state_check');
+
   return system_settings_form($form);
```

Comment:
I tested, if you make 
`$form = system_settings_form($form);`
...
`return system_settings_form($form);`
, the form is submitted twice.

`$form = system_settings_form($form);`
...
`return $form;`, do the trick.